### PR TITLE
FEAT: Like and Watch Later Functionalities

### PR DESCRIPTION
**Pull request by fahd23**
- Added Like and Watch Later Functionalities

### CODE REVIEW

**Review by JantuDeb — APPROVED** [review]
This is well written and structured code

File: src/reducer/like-reducer.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        like: [...like, payload],
+      };
+    case "REMOVE_FROM_LIKE":
+      return {
+        ...state,
+        like: [...like.filter((item) => item._id !== payload._id)],
+      };
+    case "CLEAR":
+      return payload;
+    case "default":
```
Comment (t007rushi) [inline src/reducer/like-reducer.js:23]: simply write default

File: src/context/Data-context.js [hunk 2]
```diff
@@ -1,11 +1,33 @@
-import { createContext, useContext, useState } from "react";
+import { createContext, useReducer, useContext, useState } from "react";
+import { likeReducer } from "reducer/like-reducer";
+import { watchlaterReducer } from "reducer/watchLater-reducer";
 
-const DataContext = createContext();
+const defaultValue = {
+  like: [],
+  watchlater: [],
+};
```
Comment (t007rushi) [inline src/context/Data-context.js]: no need for this, can be any placeholder null or else empty

**Review by t007rushi — APPROVED** [review]
Go ahead for merge

File: src/context/Data-context.js (hunk 2 above)
Comment (fahd23) [reply src/context/Data-context.js]: Ok Thank you

**Pull request merged**